HW04/Pavel_Mishchenko/lib/mentor.rb

```diff
@@ -0,0 +1,42 @@
+# frozen_string_literal: true
+
+class Mentor
+  include StudentNotificator
+  attr_reader :name, :surname
+  attr_accessor :notifications_list
+
+  def initialize(name:, surname:)
+    super()
+    @name = name
+    @surname = surname
+    @notifications_list = []
+  end
+
+  def add_homework(homework_parameters)
+    new_homework = Homework.new(title: homework_parameters[:title],
+                                description: homework_parameters[:description],
+                                student: homework_parameters[:student],
+                                mentor: self)
+    notify_student(homework_parameters[:student], "New homework! #{homework_parameters[:title]}")
+    new_homework
+  end
+
+  def mark_as_read!
+    notifications_list.each { |notification| notification.status = 'read' }
+  end
+
+  def reject_to_work!(homework)
+    notify_student(homework.student, 'Homework is rejected!')
+  end
+
+  def accept!(homework)
+    homework.accepted = true
+    notify_student(homework.student, 'Your homework is accepted! Good job!')
+  end
+
+  def notifications
```

Comment:
let's just return all notifications, and remove them from the business logic

Reply:
ok, you can leave it as it is, but I'm a little against displaying information on the screen in this method ( puts )